### PR TITLE
feat: save filter on local storage

### PR DESCRIPTION
**Pull request by adnanjpg**
closes https://github.com/acikkaynak/afet-destek-org/issues/171

### CODE REVIEW

File: lib/pages/demands_page/state/demands_state.dart [hunk 1]
```diff
@@ -1,21 +1,40 @@
...
+  DemandsState._();
+
+  List<String>? get categoryIds => filter?.categoryIds;
+  double? get filterRadiusKm => filter?.filterRadiusKm;
 
   bool get hasAnyFilters => categoryIds != null || filterRadiusKm != null;
 }
 
+@freezed
+@HiveType(typeId: 0)
```
Comment (easeccy) [inline lib/pages/demands_page/state/demands_state.dart:24]: Use `shared_preferences` instead.

**Review by easeccy — CHANGES_REQUESTED** [review]
Fixes needed.

File: lib/pages/demands_page/state/demands_cubit.dart [hunk 2]
```diff
@@ -27,6 +28,32 @@ class DemandsCubit extends Cubit<DemandsState> {
   int _page = 1;
   bool _isLastPage = false;
 
+  Future<SharedPreferences> get _prefs => SharedPreferences.getInstance();
+  static const String _filterKey = 'demands_filter';
```
Comment (adnanjpg) [inline lib/pages/demands_page/state/demands_cubit.dart:32]: filterKey = 'filterKey'

File: lib/pages/demands_page/state/demands_cubit.dart [hunk 3]
```diff
@@ -27,6 +28,32 @@ class DemandsCubit extends Cubit<DemandsState> {
...
+
+  Future<void> initFilterState() async {
```
Comment (adnanjpg) [inline lib/pages/demands_page/state/demands_cubit.dart:34]: convert to init, await and try catch initing filter, then fetch demands, remove getDemands

File: lib/pages/demands_page/state/demands_cubit.dart [hunk 4]
```diff
@@ -83,14 +109,26 @@ class DemandsCubit extends Cubit<DemandsState> {
...
-        filterRadiusKm: filterRadiusKm,
+        filter: DemandsStateFilter(
+          categoryIds: categoryIds,
+          filterRadiusKm: filterRadiusKm,
+        ),
         demands: null,
       ),
     );
 
+    unawaited(
```
Comment (adnanjpg) [inline lib/pages/demands_page/state/demands_cubit.dart:123]: try catch

File: lib/pages/demands_page/state/demands_state.dart [hunk 5]
```diff
@@ -1,21 +1,41 @@
...
 
 @freezed
 class DemandsState with _$DemandsState {
-  const factory DemandsState({
+  factory DemandsState({
     required DemandsStateStatus status,
     required List<Demand>? demands,
-    required List<String>? categoryIds,
-    required double? filterRadiusKm,
+    DemandsStateFilter? filter,
```
Comment (adnanjpg) [inline lib/pages/demands_page/state/demands_state.dart:12]: non nullable, required, defualt .empty

File: lib/pages/demands_page/state/demands_state.dart [hunk 6]
```diff
@@ -1,21 +1,41 @@
 import 'package:afet_destek/data/models/demand.dart';
 import 'package:freezed_annotation/freezed_annotation.dart';
 
+part 'demands_state.g.dart';
 part 'demands_state.freezed.dart';
 
 @freezed
 class DemandsState with _$DemandsState {
-  const factory DemandsState({
+  factory DemandsState({
```
Comment (adnanjpg) [inline lib/pages/demands_page/state/demands_state.dart:9]: const

File: lib/pages/demands_page/state/demands_state.dart [hunk 7]
```diff
@@ -1,21 +1,41 @@
...
+  DemandsState._();
+
+  List<String>? get categoryIds => filter?.categoryIds;
+  double? get filterRadiusKm => filter?.filterRadiusKm;
 
   bool get hasAnyFilters => categoryIds != null || filterRadiusKm != null;
 }
 
+@freezed
+class DemandsStateFilter with _$DemandsStateFilter {
```
Comment (adnanjpg) [inline lib/pages/demands_page/state/demands_state.dart:23]: const

File: lib/pages/demands_page/state/demands_cubit.dart (hunk 4 above)
Comment (easeccy) [reply lib/pages/demands_page/state/demands_cubit.dart:123]: No need, this is unawaited, won't block the next line on error. But remove the bang and make it non-nullable.

**Comment by adnanjpg** [comment]
changes done, please review @easeccy

**Review by easeccy — APPROVED** [review]

**Comment by easeccy** [comment]
Looks great, thanks.

**Pull request merged**